TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/GET-Positive.robot

```diff
@@ -1,90 +1,109 @@
+    When Query All Device Profiles With offset=2
+    Then Should Return Status Code "200" And profiles
+    And Should Return Content-Type "application/json"
+    And Should Be True  len(${content}[profiles]) == 1
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete Multiple Device Profiles By Names  Test-Profile-1  Test-Profile-2  Test-Profile-3
 
 ProfileGET003 - Query all device profiles by limit
-    Given Create Multiple Device Profiles
-    When Query All Device Profiles
-    Then Should Return Status Code "200"
-    And Should Have Content-Type "application/json"
-    And Validate Response Schema
-    And Returned Number Should Be The Same As Setting
-    And Response Time Should Be Less Than "1200"ms
-    [Teardown]  Delete Device Profiles
+    Given Generate Multiple Device Profiles Sample
+    And Create Device Profile ${deviceProfile}
+    When Query All Device Profiles With limit=2
+    Then Should Return Status Code "200" And profiles
+    And Should Return Content-Type "application/json"
+    And Should Be True  len(${content}[profiles]) == 2
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete Multiple Device Profiles By Names  Test-Profile-1  Test-Profile-2  Test-Profile-3
 
 ProfileGET004 - Query all device profiles by labels
-    Given Create Multiple Device Profiles
-    When Query All Device Profiles
-    Then Should Return Status Code "200"
-    And Should Have Content-Type "application/json"
-    And Validate Response Schema
-    And Profiles Should Be Linked To Specified Label
-    And Response Time Should Be Less Than "1200"ms
-    [Teardown]  Delete Device Profiles
+    Given Generate Multiple Device Profiles Sample
+    And Create Device Profile ${deviceProfile}
+    When Query All Device Profiles With labels=bacnet
+    Then Should Return Status Code "200" And profiles
+    And Should Return Content-Type "application/json"
+    And Should Be True  len(${content}[profiles]) == 2
+    And Profiles Should Be Linked To Specified Labels: bacnet
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete Multiple Device Profiles By Names  Test-Profile-1  Test-Profile-2  Test-Profile-3
 
 ProfileGET005 - Query device profile by name
-    Given Create A Device Profile
-    When Query Device Profile By Name
-    Then Should Return Status Code "200"
-    And Should Have Content-Type "application/json"
-    And Validate Response Schema
-    And Response Time Should Be Less Than "1200"ms
-    [Teardown]  Delete Device Profiles
+    Given Generate An Device Profile Sample  Test-Profile-1
+    And Create Device Profile ${deviceProfile}
+    When Query Device Profile By Name  Test-Profile-1
+    Then Should Return Status Code "200" And profile
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
```

Comment:
Suggest validating the response contains the correct profile name.

Reply:
Added the validations to device profile and devices ervice
https://github.com/edgexfoundry/edgex-taf/blob/ead9e9a1d381178856840a022e04d9d054862201/TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceservice/GET-Positive.robot#L67
https://github.com/edgexfoundry/edgex-taf/blob/ead9e9a1d381178856840a022e04d9d054862201/TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/GET-Positive.robot#L61